src/tiledb/cloud/bioimg/ingestion.py

```diff
@@ -386,7 +386,8 @@ def register_dataset_udf(
         )
     if compute:
         run_dag(graph, debug=verbose)
-    return graph
+    else:
+        return graph
```

Comment:
@ktsitsi to sum things up: a graph that has been run can't be returned, but a graph that hasn't been run can be pickled and returned. Is that right?

I'm uncomfortable with this if/else from an API design perspective. I may be missing something, but it looks like unnecessary complexity. Could we do the following instead when we want to run the graph?

```python
graph = register_dataset_udf(...)
run_dag(graph)
```

It's less code and has less ambiguity about intent.

Reply:
@sgillies I believe the problem is there are some components that get created during execution, i.e a monitor thread: https://github.com/TileDB-Inc/TileDB-Cloud-Py/blob/9a14c4e9710ee9d60e0d3f46fdd40eddc3ad49d7/src/tiledb/cloud/dag/dag.py#L1173 . So a graph that is running or has completed might contain components that cloudpickle can't handle.

A separate work item should be to cleanup these classes so we don't need to differentiate this.